packages/remirror__extension-list/src/task-list-extension.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  ApplySchemaAttributes,
+  assertGet,
+  command,
+  CommandFunction,
+  extension,
+  ExtensionPriority,
+  ExtensionTag,
+  getMatchString,
+  InputRule,
+  keyBinding,
+  KeyBindingProps,
+  NamedShortcut,
+  NodeExtension,
+  NodeExtensionSpec,
+  nodeInputRule,
+  NodeSpecOverride,
+  NodeViewMethod,
+  ProsemirrorNode,
+} from '@remirror/core';
+import { ExtensionListMessages as Messages } from '@remirror/messages';
+import { NodeSelection, TextSelection } from '@remirror/pm/state';
+import { ExtensionListTheme } from '@remirror/theme';
+
+import { toggleList } from './list-commands';
+import { TaskListItemExtension } from './task-list-item-extension';
+
+/**
+ * Create the node for a task list.
+ */
+@extension<object>({
+  defaultOptions: { priority: ExtensionPriority.Medium },
+})
+export class TaskListExtension extends NodeExtension {
+  get name() {
+    return 'taskList' as const;
+  }
+
+  createTags() {
+    return [ExtensionTag.Block, ExtensionTag.ListContainerNode];
+  }
+
+  createNodeSpec(extra: ApplySchemaAttributes, override: NodeSpecOverride): NodeExtensionSpec {
+    return {
+      content: 'taskListItem+',
+      ...override,
+      attrs: extra.defaults(),
+      parseDOM: [
+        { tag: 'ul[data-task-list]', getAttrs: extra.parse },
+        ...(override.parseDOM ?? []),
+      ],
+      toDOM: (node) => ['ul', { ...extra.dom(node), 'data-task-list': '' }, 0],
+    };
+  }
+
+  createNodeViews(): NodeViewMethod | Record<string, never> {
+    return (_, view, getPos) => {
+      const dom = document.createElement('ul');
+      dom.style.position = 'relative';
+
+      const pos = (getPos as () => number)();
+      const $pos = view.state.doc.resolve(pos + 1);
+
+      const parentListItemNode: ProsemirrorNode | undefined = $pos.node($pos.depth - 1);
+
+      const isFirstLevel = parentListItemNode?.type?.name !== 'listItem';
+
+      if (!isFirstLevel) {
+        const parentListItemPos: number = $pos.start($pos.depth - 1);
+
+        const spine = document.createElement('div');
+        spine.contentEditable = 'false';
+        spine.classList.add(ExtensionListTheme.LIST_SPINE);
+
+        spine.addEventListener('click', (event) => {
+          const selection = NodeSelection.create(view.state.doc, parentListItemPos - 1);
+          view.dispatch(view.state.tr.setSelection(selection));
+          this.store.commands.toggleListItemClosed();
+
+          event.preventDefault();
+          event.stopPropagation();
+        });
+        dom.append(spine);
+      }
+
+      const contentDOM = document.createElement('div');
+      contentDOM.classList.add(ExtensionListTheme.UL_LIST_CONTENT);
+      dom.append(contentDOM);
+
+      return {
+        dom,
+        contentDOM,
+      };
+    };
+  }
+
+  createExtensions() {
+    return [
+      new TaskListItemExtension({
+        // The priority is `Medium` instead of `Low` because we want `TaskListItemExtension` has
```

Comment:
Just wondering if you meant default here rather than low? I believe default is the normal priority?

```suggestion
        // The priority is `Medium` instead of `Default` because we want `TaskListItemExtension` to have a
```